src/03-modality-agnostic-files.md

```diff
@@ -452,8 +452,13 @@ imaging acquisition sequence (each *run* file) within one session.
 
 Each neural recording *file* SHOULD be described by exactly one row.
 Some recordings consist of multiple parts, that span several files,
-for example through `echo-`, `part-`, or `split-` entities.
-Such recordings MUST be documented with one row per file.
+for example through `echo-`, `part-`, or `split-` entities,
+or due to the file format such as in the BrainVision (`.vhdr`, `.vmrk`, `.eeg`)
+or EEGLAB (`.set`, `.fdt`) formats.
+Recordings with multiple parts MUST be documented with one row per file.
+It is RECOMMENDED that if the multiple parts of a recording stem solely from the file format,
+that file is listed, which would also be passed to software functions for
+reading (for example `.vhdr` for BrainVision and `.set` for EEGLAB).
```

Comment:
I think the wording needs to be adjusted. In the previous sentence, file formats with multiple files are treated the same as [file collections](https://bids-specification.readthedocs.io/en/stable/02-common-principles.html#entity-linked-file-collections), in which each file MUST be documented separately in the scans.tsv. However, in the next sentence, it is RECOMMENDED to _not_ do that for these file formats specifically.

I think that you need to distinguish between file collections and multi-file file formats from the beginning, so that it's clear that multi-file format files _shouldn't_ be listed separately in the scans.tsv file.

Reply:
I actually wonder whether it's not better to be consistent and also document multi-file file formats with one row **per file** then. That is, BrainVision recordings would have 3 rows per recording in `scans.tsv`, and EEGLAB recordings that make use of their optional `.fdt` file get 2 rows per recording in `scans.tsv`.

It's less pretty than just listing the header file there (`.vhdr` for BrainVision, `.set` for EEGLAB), but not sure if it's worth introducing a special rule then :thinking: As far as I know we don't have other multi-file file formats (excluding those that are directories, .. here the whole directory gets listed)

What do you think @hoechenberger ?

Reply:
My proposal would be to allow `filename` to be an array of strings (pity that it's `scans.tsv` and not `scans.json`!!), this would ensure we still have only one row per scan. Still – which file should we try to load? And what about split files?

-> Considering this, I'm leaning toward listing only the VHDR file for BV and SET file for EEGLAB.

But there shouldn't be a RECOMMENDED, it should be a MUST, and we shouldn't just propose, but enforce which file to list. I don't like all this "well yeah umm you could but you don't have do, do what you like" stuff in the standard, it makes software implementations such an unnecessary pain!

Reply:
> Still – which file should we try to load?

That is less of a problem than it seems because all data formats and their extensions are documented in BIDS and it's easy to include a logic like:

`"if VMRK found, this is BrainVision, so look for VHDR, if it doesn't exist, raise error"`

> And what about split files?

Here we actually recommend that people write ALL split files, see:

- https://bids-specification.readthedocs.io/en/latest/99-appendices/14-glossary.html#split-entities
- implemented in https://github.com/bids-standard/bids-specification/pull/735
- (after mne-bids discussion in https://github.com/mne-tools/mne-bids/issues/692)

So that further makes me think that listing **all** neural recordings in scans.tsv makes sense ... also for the two exceptions we have: EEGLAB (set+fdt) and BrainVision (vhdr, vmrk, eeg) files :thinking: 

> But there shouldn't be a RECOMMENDED, it should be a MUST, and we shouldn't just propose, but enforce which file to list.

right, but this is the age-old discussion about backward compatibility and the balance we are treading ... I think things will get better in the long run with the move to the BIDS schema, which will eventually allow validation of BIDS datasets against particular BIDS schema versions (something we cannot currently do with bids-validator) --> and I think that may make us all more comfortable with breaking changes and including rules that are more strict.

(as you know I do in principle agree with you that we need more MUST and less SHOULD)

**TL;DR: So I think I'll revamp this PR so that **all** BrainVision files and **all** EEGLAB files SHOULD be listed in scans.tsv**

Reply:
> TL;DR: So I think I'll revamp this PR so that all BrainVision files and all EEGLAB files SHOULD be listed in scans.tsv

I really don't think this is a good idea. It effectively creates duplicates for those scans. If you really want to list all files, we should find a way to squeeze ONR recording into ONE row, and not spread it out across several rows.

Reply:
> It effectively creates duplicates for those scans.

not really duplicates (the extension is different) ... one could argue that it shows that several files belong to that recording. After all, we document that each file belonging to a recording gets an entry in the table.

Also, we are already doing this for split files, and for qMRI file collections

I think it'd be the consistent thing to do ... and I think it's kind of implicit in the present rules already: just not obviously documented (which I'd like to fix).

Reply:
> not really duplicates (the extension is different)

Yes but that is exactly my point :) the purpose of scans.tsv is, and I state the spec:

> The purpose of this file is to describe timing and other properties of each imaging acquisition sequence (each run file) within one session. Each neural recording file SHOULD be described by exactly one row.

So the intention was, originally, to have precisely one row per acquisition run, from where you could extract all the relevant metadata for that specific run, such as recording time and yes, also the name of file used to store the data. 

Later on we extended the standard, and I believe I contributed or at least +1'd this change as well:
> Some recordings consist of multiple parts, that span several files, for example through echo-, part-, or split- entities. Such recordings MUST be documented with one row per file.

And I believe this was a mistake. As users are free to add additional columns to the file, we end up with duplicated metadata in cases of split, ... files.

If the focus and purpose of this file really has shifted that much since its initial inception, it should be renamed to files.tsv or something. Instead of squeezing more and more new stuff into a thing that was never designed to store that kind of info, we should just add a new file. 

I know there's basically no chance to make this happen, but I wanted to voice that I'm very unhappy with the direction some things are developing. 😓

Reply:
.... but things are a total mess already anyway ... yeah, it's probably more consistent to add those additional files than just omitting them.

Reply:
> I know there's basically no chance to make this happen, but I wanted to voice that I'm very unhappy with the direction some things are developing. sweat

knowing your opinions to some extent I don't think this statement does the current developments justice ... the present issue is more "cleaning up" / "documenting" rather than developing anyhow. But maybe we should discuss that in person some other time :-)